src/FactoryPieces.cpp

```diff
@@ -0,0 +1,15 @@
+#include "FactoryPieces.h"
+
+unique_ptr<Piece> FactoryPieces::create(const char& _ch, const Location& loc, PlayerColor color, Board* board)
+{
+	auto it = FactoryPieces::getPiecesMap().find(_ch);

```

Comment:
One potential issue with this code is that the getPiecesMap() function is called twice in FactoryPieces::create(), which could be inefficient if the map is large. It would be better to call it once and store the result in a local variable.